app/java/net/openid/appauthdemo/IdentityProvider.java

```diff
@@ -219,6 +220,12 @@ public void setClientId(String clientId) {
         mClientId = clientId;
     }
 
+    public String getClientSecret() {
+        return mClientSecret;
+    }
+
+    public void setClientSecret(String clientSecret) { mClientSecret = clientSecret; }
```

Comment:
Native apps should not have client secrets unless they are provisioned via dynamic client registration.  I generally oppose security theatre as it leads people not to take the real precautions.